### PR TITLE
[DRAFT] Refactor user creation related code

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
The goal of this pull-request is to refactor the code related to users
* User creation
* User password management

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -548,43 +577,49 @@ def _connect_mongodb_exporter(self) -> None:
...
-            return
-        event.set_results({"password": self.get_secret("app", f"{username}_password")})
+            return ""
+        return username
+
+    def _on_get_password(self, event: ActionEvent) -> None:
+        """Returns the password for the user as an action response."""
+        username = self._get_user_or_fail(event)
+        password = self.get_secret("app", get_password_key_name_for_user(username))
+        logger.error(f"_on_get_password Password for {username} is {password}")
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply src/charm.py:621]: That was for test debug purposes,  removed

File: lib/charms/mongodb/v0/users.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+LIBPATCH = 1
+
+
+class OperatorUser:
+    """Operator user for MongoDB."""
+
+    username = "operator"
+    password_key = f"{username}_password"
+    database = "admin"
+    role = "default"
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply lib/charms/mongodb/v0/users.py:22]: Done

File: lib/charms/mongodb/v0/users.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class MonitorUser:
+    """Monitor user for MongoDB."""
+
+    username = "monitor"
+    password_key = f"{username}_password"
+    database = ""
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply lib/charms/mongodb/v0/users.py:30]: Done

File: src/charm.py [hunk 4]
```diff
@@ -593,22 +628,12 @@ def _on_set_password(self, event: ActionEvent) -> None:
...
-            event.fail(
-                f"The action can be run only for users used by the charm: {CHARM_USERS} not {username}."
-            )
-            return
-
-        new_password = generate_password()
-        if "password" in event.params:
-            new_password = event.params["password"]
+        username = self._get_user_or_fail(event)
+        new_password = event.params.get(EVENT_PASSWORD_PARAM, generate_password())
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply src/charm.py:632]: Fixed, thank you

**Review by delgod — APPROVED** [review]

**Pull request closed without merging**